aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -222,15 +222,11 @@ TORCH_META_FUNC(triangular_solve)(const Tensor& self, const Tensor& A, bool uppe
     auto nrows = A.size(-2);
 
     // make column major strides for BLAS
-    auto solution_strides = at::detail::defaultStrides(self_broadcast_size);
-    solution_strides[ndim - 2] = 1;
-    solution_strides[ndim - 1] = nrows;
+    const auto solution_strides = at::native::contiguous_strides(self_broadcast_size, /*f-contig=*/true);
```

Comment:
Thanks for fixing that!
Why the tests were not failing? You mentioned that autograd doesn't like matrices with 0 strides and `sample_inputs_linalg_solve` from `common_methods_invocations.py` has 0x0 input matrix case.

Reply:
It's forward AD that was failing. Can it be the case that this function does not support forward AD?